docs/API.md

```diff
@@ -55,6 +55,10 @@ If the request object is invalid JSON, an error with code `-32700` is reported.
 
 If no method is specified or an invalid method is specified, an error with code `-32601` is reported.
 
+If no `params` object is specified, or the `param` object for the specified
+method is invalid, an error with code `-32602` is reported. For more information
+on the validation error data format see [Validation error data].
```

Comment:
How does this match API methods that are defined to return JSON without any param object?

Reply:
From what i have seen, all methods need a params object, if no parameters are needed then this object is empty, as it is the case with `version_info` for example:

```perl
$json_schemas{version_info} = joi->object->strict;
```

Reply:
That is not how it is defined in the API doucmentation, e.g. https://github.com/zonemaster/zonemaster-backend/blob/develop/docs/API.md#api-method-version_info

Reply:
My mistake, this schema is defined but never used as the params object is only validated if the schema has at least one property marked as required (I am not sure this is the right thing to do, a methods with only optional parameters would never have its request validated).